### ISSUE TITLE
Option Missing from the Documentation

### ISSUE THREAD

**Issue by bsampica** [chkr1011/MQTTnet#297]
Christian, can you explain the purpose of this option?

.WithReceivedApplicationMessageProcessingMode(MqttReceivedApplicationMessageProcessingMode.DedicatedThread)

With a possible Enum option of MqttReceivedApplicationMEssageProcessingMode.SingleThread | DedicatedThread

And does it relate to the managed client as well?

**Comment by chkr1011** [comment]
Hi,
this option defines how the incoming messages are processed.
# SingleThread
The messages are processed in serial order after each other. One long performing operation will slow down every following messages.

# DedicatedThread
A new (ThreadPool) thread is started for every incoming message. With this option a long running message will **not** slow down other messages but the order is lost and there is some overhead in creating the threads. Also they are processed in parallel what gains some performance improvement (depending on use case).

I will add this later to the Wiki. This also applies for the managed client because it also uses a _regular_ client inside.

Best regards
Christian

**Comment by bsampica** [comment]
Hey, thanks for the quick response @chkr1011 - what is the default setting if no options are specified?

**Comment by chkr1011** [comment]
The setting is removed now because it causes some new issues. By default all incoming messages are processed in the worker thread. If processing a app message took longer you need to start a new Task on your own in the event handler.

**Comment by bsampica** [comment]
Ok. That actually works better for me, because my mqtt client is in a asp.net<http://asp.net> core webapp and is a subscriber...

The whole client is hosted as a micro service which is already a background task

On Jun 4, 2018, at 3:15 PM, Christian <notifications@github.com<mailto:notifications@github.com>> wrote:


The setting is removed now because it causes some new issues. By default all incoming messages are processed in the worker thread. If processing a app message took longer you need to start a new Task on your own in the event handler.
